library/Makefile

```diff
@@ -1,8 +1,21 @@
 
 # Also see "include/mbedtls/config.h"
 
+ifeq ($(CC),iccarm)
+CFLAGS	?= --arm -Om --silent
+
+# gcc comparable warning flags are not present in IAR.
+# All warnings are enabled by default.
+WARNING_CFLAGS ?=
+else
 CFLAGS	?= -O2
 WARNING_CFLAGS ?=  -Wall -W -Wdeclaration-after-statement
+endif
+
+# IAR archive command does not have ar compatible options 
+ifneq ($(AR),iarchive)
```

Comment:
> Theoretically yes but mostly we add tools to the path

This is not correct in the case of mbed TLS. In fact, most of our scripts are driven through environment variables. In some cases, this is even mandatory. For example, in the case of ARMC5 and ARMC6 we cannot have them both in the path simultaneously, so we actually need to pass full directory locations to the `all.sh` script.